dpytools/http/dataset/base_dataset.py

```diff
@@ -0,0 +1,71 @@
+import json
+from typing import Dict
+
+from requests import RequestException, Response
+
+from dpytools.http.base import BaseHttpClient
+from dpytools.http.upload.token_auth import TokenAuth
+from dpytools.logging.logger import DpLogger
+
+logger = DpLogger("dpytools")
+
+
+class BaseDatasetClient(BaseHttpClient):
+    def __init__(self, url: str, backoff_max: int = 30):
+        super().__init__(backoff_max=backoff_max)
+        self.token_auth = TokenAuth(backoff_max=backoff_max)
+        self.url = url
+        self.dataset_url = f"{self.url}/dataset"
+
+    def send_json(self, url: str, json_data: Dict) -> None:
+        """
+        Send a POST request with JSON data to the specified URL.
+
+        :param url: The URL to send the POST request to.
+        :param json_data: The JSON data to include in the POST request.
+        """
+        try:
+            response = self.post(
+                url,
+                headers=self.token_auth.get_auth_header(),
+                json=json_data,
+                verify=True,
+            )
+            response.raise_for_status()
+            return response
+        except RequestException as e:
+            logger.error("Failed to send POST request", exc_info=e)
+        raise
+
+    def _assign(self, upload_dict: Dict) -> None:
+        """
+        Assign the upload dictionary to the instance variable.
+
+        :param upload_dict: Dictionary containing upload data.
+        """
+        if not hasattr(self, "upload_dict"):
+            self.upload_dict = upload_dict
+        else:
+            logger.warning("upload_dict is already assigned.")
+
+    def _get_recipe_id(self, dataset_id: str) -> None:
```

Comment:
Personally I think this whole method should be removed. The recipe API is a separate API to the dataset API and it is used for CMD and cantabular datasets, I don't think our stuff will be on the recipe API at all. Also the end point you have used to the recipe API is incorrect (it should be /recipes rather than /dataset/recipes)